datadog_lambda/wrapper.py

```diff
@@ -276,6 +300,20 @@ def _after(self, event, context):
                 else:
                     self.inferred_span.finish()
 
+            if should_trace_cold_start:
+                try:
+                    following_span = self.span or self.inferred_span
```

Comment:
Hmmmm, is this order correct? I think we'd want to prefer the inferred span if it exists, right?

Although it may not matter as the extension should reparent the span anyway, as I see in this screenshot:
<img width="941" alt="image" src="https://user-images.githubusercontent.com/1598537/218536922-cda4365d-2717-492a-b8e5-7e787cbae80b.png">

Reply:
Updated. Good catch!

Reply:
Actually, I just tested `following_span = self.inferred_span or self.span` and because the `inferred_span` can be the root_span, it would make the `aws.lambda.load`'s finish_time < start_time and become a 0 length span.
[SPAN LINK](https://ddserverless.datadoghq.com/apm/trace/4685652051164053004)
<img width="924" alt="Screenshot 2023-02-13 at 3 30 08 PM" src="https://user-images.githubusercontent.com/5253430/218568041-182d7162-c82a-4188-99a6-0a705ebf5c95.png">

Reply:
Ahh! That's completely my mistake, I misunderstood what this `following_span` was being used for, I'm sorry I distracted you.